contracts/staking/TribalChief.sol

```diff
@@ -114,11 +120,24 @@ contract TribalChief is CoreRef, ReentrancyGuard {
     event NewTribePerBlock(uint256 indexed amount);
     event PoolLocked(bool indexed locked, uint256 indexed pid);
 
+    /// @notice The way this function is constructed, you will not be able to 
+    /// call initialize after this function is constructed, effectively 
+    /// only allowing TribalChief to be used through delegate calls.
+    /// @param coreAddress The Core contract address.
+    constructor(address coreAddress) CoreRef(coreAddress) {}
```

Comment:
I have thought so much about proxy patterns that I believe it'd be great for us to talk more at length about this, @Joeysantoro . 😄 🙌

Reply:
Definitely